scripts/pcvGuardianDeploy.ts

```diff
@@ -0,0 +1,88 @@
+import hre, { ethers } from 'hardhat';
+import { expect } from 'chai';
+import config from './config';
+import { Core, PCVGuardian } from '@custom-types/contracts';
+import { getImpersonatedSigner } from '@test/helpers';
+
+const {
+  CORE,
+  PROTOCOL_MULTISIG_ADDRESS,
+  VOLT_FUSE_PCV_DEPOSIT,
+  PCV_DEPOSIT,
+  PRICE_BOUND_PSM,
+  PCV_GUARD_EOA_1,
+  PCV_GUARD_EOA_2
+} = config;
+
+const PCV_GUARD_ROLE = ethers.utils.id('PCV_GUARD_ROLE');
+const PCV_GUARD_ADMIN_ROLE = ethers.utils.id('PCV_GUARD_ADMIN_ROLE');
+
+async function deploy() {
+  const PCVGuardian = await ethers.getContractFactory('PCVGuardian');
+  const whitelistAddresses = [VOLT_FUSE_PCV_DEPOSIT, PCV_DEPOSIT, PRICE_BOUND_PSM];
+
+  const pcvGuardian = await PCVGuardian.deploy(CORE, PROTOCOL_MULTISIG_ADDRESS, whitelistAddresses);
+  await pcvGuardian.deployed();
+
+  // Deploy PCV Guardian
+  console.log('\n ~~~~~ Deployed PCV Guardian Successfully ~~~~~ \n');
+  console.log(`PCV Guardian:        ${pcvGuardian.address}`);
+
+  const PCVGuardAdmin = await ethers.getContractFactory('PCVGuardAdmin');
+  const pcvGuardAdmin = await PCVGuardAdmin.deploy(CORE);
+  await pcvGuardAdmin.deployed();
+
+  //Deploy PCV Guard Admin
+  console.log('\n ~~~~~ Deployed PCV Guard Admin Successfully ~~~~~ \n');
+  console.log(`PCV Guard Admin:        ${pcvGuardAdmin.address}`);
+
+  if (hre.network.name !== 'mainnet') {
+    const core = await ethers.getContractAt('Core', CORE);
+
+    const signer = await getImpersonatedSigner(CORE);
```

Comment:
Let's simulate with the multisig here, I know I said we could use core, but we should simulate with the exact same conditions that we have on mainnet